server/src/main/java/pl/edu/agh/RaftServer.java

```diff
@@ -49,23 +52,28 @@
     private static final ScheduledExecutorService TIMEOUT_EXECUTOR = Executors.newSingleThreadScheduledExecutor();
 
     private final SocketAddress localAddress;
-    private final TcpServer<ByteBuf, ByteBuf> tcpServer;
-    private final Map<SocketAddress, Connection<ByteBuf, ByteBuf>> serverConnections;
+    private Map<SocketAddress, Connection<ByteBuf, ByteBuf>> serverConnections = Maps.newHashMap();
```

Comment:
From Java 7 the diamond syntax is preferred over the factory methods from Guava. Moreover, it shouldn't be initialized here because it is initialized in the constructor. It should also remain final.